internal/driver/config/namespace_watcher.go

```diff
@@ -22,9 +22,20 @@ import (
 )
 
 type (
+	parser func([]byte, interface{}) error
+
+	NamespaceFile struct {
+		Name     string
+		Contents []byte
+		Parser   parser
+
+		err       error
```

Comment:
Yep, on re-read, the err is not necessary to keep around. I'll remove and add a comment that states that the namespace will be nil if it failed to parse

Reply:
Ah ok I see, maybe it should rather keep the old config if the change update failed. That way nothing breaks while running. In that case it would make sense to have an error and as content the previously successful config. I am just curious where this error is handled in the end.

Reply:
I see what you mean- that's a good strategy. What I can do is if a namespace fails to parse, I'll update the `Contents` (maybe re-name to `LastReadContents`) but keep the previously parsed namespace.